### PR TITLE
2016Q4.vanishingpoint-copyover

### PR DESCRIPTION
**Pull request by crispin-kirchner**

### CODE REVIEW

**Comment by bjornborg** [comment]
Shouldn't this be PR into pre-release?

**Comment by crispin-kirchner** [comment]
@bjornborg thanks for pointing this out, I forgot

When I try to change the base (via `edit` up on this page) I get the following message:

```
Some commits from the old base branch may be removed from the timeline, and old review comments may become outdated.
```

I don't understand fully what's meant and therefore I'm hesitant to just do it - any experiences with that?

**Comment by bjornborg** [comment]
I got your back.

**Comment by crispin-kirchner** [comment]
thanks ;)

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp [hunk 1]
```diff
@@ -0,0 +1,283 @@
+#include <cstdlib>
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: Please add the default License header and your name as author.

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    LMVP_UNUSED(userdata);
+
+    if(event == cv::EVENT_MOUSEMOVE) {
+        std::cout << "x: " << x << " y: " << y << std::endl;
+    }
+}
+
+class VanishingPointLowPass {
+public:
+    VanishingPointLowPass(float dampingFactor) : dampingFactor_(dampingFactor), lastVanishingPoint_() { }
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: The argument should be changed to "const float &dampingFactor".

File: code/system/application/legacy/vanishingpoint/apps/ImageProcessingTestFixture.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#include <iostream>
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/ImageProcessingTestFixture.cpp:24]: Please add a short License header and a few lines of explanation for the purpose of this file.

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+void onMouse(int event, int x, int y, int flags, void * userdata) {
+    LMVP_UNUSED(flags);
+    LMVP_UNUSED(userdata);
+
+    if(event == cv::EVENT_MOUSEMOVE) {
+        std::cout << "x: " << x << " y: " << y << std::endl;
+    }
+}
+
+class VanishingPointLowPass {
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: Please separate the classes into a header and implementation file.

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    FailedAttachingToSharedMemoryException(const std::string & name)
+        : std::runtime_error("Failed attaching to shared memory " + name) {
+    }
+};
+
+/**
+ * Class linking a SharedImage to a cv::Mat. Attachment to the shared memory happens on construction.
+ * Construction only succeeds when attaching to the shared memory was successful.
+ */
+class SharedImageCvAdapter {
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: Please separate the classes into a header and implementation file.

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        imageHeader_ = cv::Mat(sharedImage.getHeight(), sharedImage.getWidth(),
+                               CV_MAKETYPE(CV_8U, sharedImage.getBytesPerPixel()),
+                               sharedImageMemory_->getSharedMemory());
+    }
+
+    std::shared_ptr<SharedMemory> sharedImageMemory_;
+    cv::Mat imageHeader_;
+};
+
+class LaneMarkingsVanishingPointDetectionModule : public DataTriggeredConferenceClientModule {
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: Please separate the classes into a header and implementation file (separate header and implementation files for the individual classes).

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+            }
+            catch(const FailedAttachingToSharedMemoryException & e) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+#ifndef NDEBUG
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: Please let this function be part of the regular source tree and use if based on the value of your DEBUG_IS_SET value.

File: code/system/application/legacy/vanishingpoint/include/ConfigDeserialization.h [hunk 8]
```diff
@@ -0,0 +1,129 @@
+#ifndef CONFIG_DESERIALIZATION_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/ConfigDeserialization.h]: Please add the License header and your name as author.

File: code/system/application/legacy/vanishingpoint/include/ImagePreprocessing.h [hunk 9]
```diff
@@ -0,0 +1,56 @@
+#ifndef IMAGE_PREPROCESSING_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/ImagePreprocessing.h]: Please add a License header and your name as author.

File: code/system/application/legacy/vanishingpoint/include/ImagePreprocessing.h [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+class ImagePreprocessing {
+public:
+    virtual ~ImagePreprocessing() {}
+
+    virtual cv::Mat preprocessImage(cv::Mat &) const = 0;
+};
+
+class AstaZeroImagePreprocessing : public ImagePreprocessing {
+public:
+    AstaZeroImagePreprocessing(int blurSize, uint8_t threshold) : blurSize_(blurSize), threshold_(threshold) {
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/ImagePreprocessing.h]: Please have the implementation in separate implementation files.

File: code/system/application/legacy/vanishingpoint/include/ImagePreprocessing.h [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+    }
+
+    cv::Mat preprocessImage(cv::Mat & coloredImage) const {
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/ImagePreprocessing.h]: Same here.

File: code/system/application/legacy/vanishingpoint/include/LaneMarkingsScan.h [hunk 12]
```diff
@@ -0,0 +1,373 @@
+#ifndef LANE_MARKINGS_SCAN_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/LaneMarkingsScan.h]: Please add a License header and your name.

File: code/system/application/legacy/vanishingpoint/include/LaneMarkingsScan.h [hunk 13]
```diff
@@ -0,0 +1,373 @@
...
+    /**
+     * @param scanDirection In which direction the scanner shall look for white pixels
+     * @param scanRegion The region in which to look for lane markings
+     * @param maxMarkingWidth After which maximum scan length per row the scanning shall be interrupted
+     * @param maxMarkingRowCount The maximum number of rows a marking shall span
+     * @param maxMarkingInclination The maximum inclination a road marking is expected to have,
+     *                              given in the amount of pixels to travel to the right/left from
+     *                              the first lane marking pixel when going to the next row
+     */
+    LaneMarkingsScanner(RelativeDirection scanDirection, const ScanRegion & scanRegion,
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/LaneMarkingsScan.h:41]: Please have the implementation in separate implementation files (please apply this for all further occurrences).

File: code/system/application/legacy/vanishingpoint/include/MatVisitInterface.h [hunk 14]
```diff
@@ -0,0 +1,274 @@
+#ifndef MAT_VISIT_INTERFACE_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/MatVisitInterface.h:1]: Cf above.

File: code/system/application/legacy/vanishingpoint/include/RegionOfInterestGeometry.h [hunk 15]
```diff
@@ -0,0 +1,34 @@
+#ifndef REGION_OF_INTEREST_GEOMETRY_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/RegionOfInterestGeometry.h:1]: Cf. above.

File: code/system/application/legacy/vanishingpoint/include/ScanRegion.h [hunk 16]
```diff
@@ -0,0 +1,269 @@
+#ifndef LANE_MARKINGS_SCAN_REGION_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/ScanRegion.h:1]: Cf. above.

File: code/system/application/legacy/vanishingpoint/include/VanishingPointDetection.h [hunk 17]
```diff
@@ -0,0 +1,268 @@
+#ifndef LANE_MARKINGS_DETECTION_H_
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/VanishingPointDetection.h]: Cf. above.

Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/include/VanishingPointDetection.h]: Once you have adjusted the sources, please run https://github.com/chalmers-revere/opendlv/blob/master/scripts/refactorCWD.sh on your sub-tree to let clang reformat your sources to follow our coding style.

File: code/system/application/legacy/vanishingpoint/src/LmvpTypes.cpp [hunk 18]
```diff
@@ -0,0 +1,18 @@
+#include "LmvpTypes.h"
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/src/LmvpTypes.cpp:1]: Cf. above.

File: code/system/application/legacy/vanishingpoint/src/MatVisitInterface.cpp [hunk 19]
```diff
@@ -0,0 +1,49 @@
+#include "MatVisitInterface.h"
```
Comment (chrberger) [inline code/system/application/legacy/vanishingpoint/src/MatVisitInterface.cpp:1]: Cf. above.

**Review by chrberger — CHANGES_REQUESTED** [review]
Please remove all images where license plates are readable.

File: code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp (hunk 2 above)
Comment (crispin-kirchner) [reply code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp]: Thank you for the review. What is our rationale behind passing primitive data types by reference?

**Comment by bjornborg** [comment]
Ping @crispin-kirchner Status update?

**Comment by crispin-kirchner** [comment]
The code in this branch is based on the CaroloCup algorithm and has become oblosete by the Hough-based implementation I made later. Shall I delete the branch?

**Comment by bjornborg** [comment]
Since it's absolete, I've closed the branch and PR.

**Pull request closed without merging**